x-pack/plugins/observability_solution/slo/public/embeddable/slo/alerts/slo_alerts_embeddable_factory.tsx

```diff
@@ -0,0 +1,133 @@
+import { Router } from '@kbn/shared-ux-router';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import {
+  initializeTitles,
+  useBatchedPublishingSubjects,
+  fetch$,
+  FetchContext,
+  useFetchContext,
+} from '@kbn/presentation-publishing';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { createBrowserHistory } from 'history';
+import { Storage } from '@kbn/kibana-utils-plugin/public';
+import { SLO_ALERTS_EMBEDDABLE_ID } from './constants';
+import { SloEmbeddableDeps, SloAlertsEmbeddableState, SloAlertsApi } from './types';
+import { SloAlertsWrapper } from './slo_alerts_wrapper';
+const history = createBrowserHistory();
+const queryClient = new QueryClient();
+
+export const getAlertsPanelTitle = () =>
+  i18n.translate('xpack.slo.sloAlertsEmbeddable.displayTitle', {
+    defaultMessage: 'SLO Alerts',
+  });
+
+export function getAlertsEmbeddableFactory(deps: SloEmbeddableDeps, kibanaVersion: string) {
+  const factory: ReactEmbeddableFactory<SloAlertsEmbeddableState, SloAlertsApi> = {
+    type: SLO_ALERTS_EMBEDDABLE_ID,
+    deserializeState: (state) => {
+      return state.rawState as SloAlertsEmbeddableState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const defaultTitle$ = new BehaviorSubject<string | undefined>(getAlertsPanelTitle());
+      const slos$ = new BehaviorSubject(state.slos);
+      const showAllGroupByInstances$ = new BehaviorSubject(state.showAllGroupByInstances);
+      const reload$ = new Subject<FetchContext>();
+      const api = buildApi(
+        {
+          ...titlesApi,
+          defaultPanelTitle: defaultTitle$,
+          serializeState: () => {
+            return {
+              rawState: {
+                ...serializeTitles(),
+                slos: slos$.getValue(),
+                showAllGroupByInstances: showAllGroupByInstances$.getValue(),
+              },
+            };
+          },
+          getSloAlertsConfig: () => {
+            return {
+              slos: slos$.getValue(),
+              showAllGroupByInstances: showAllGroupByInstances$.getValue(),
+            };
+          },
+          updateSloAlertsConfig: (update) => {
+            slos$.next(update.slos);
+            showAllGroupByInstances$.next(update.showAllGroupByInstances);
+          },
+        },
+        {
+          slos: [slos$, (value) => slos$.next(value)],
+          showAllGroupByInstances: [
+            showAllGroupByInstances$,
+            (value) => showAllGroupByInstances$.next(value),
+          ],
+          ...titleComparators,
+        }
+      );
+
+      const fetchSubscription = fetch$(api)
+        .pipe()
+        .subscribe((next) => {
+          reload$.next(next);
```

Comment:
Does this ever cause race conditions when used alongside `useFetchContext`? Ideally I assume you'd want the fetch context to update _before_ this subscription fires so you can be sure you're always querying with the latest fetch context, but just from the code it looks like this might fire first and start a reload with the _last_ fetch context instead.